File: S1/FactoriaMontana.java
package S1;

public class FactoriaMontana extends FactoriaCarreraYBicicleta {

	@Override
	public Bicicleta crearBicicleta() {
		return new BiciMontana();
	}

	@Override
	public Carrera crearCarrera() {
		return new CarreraMontana();
	}

}

File: S1-voluntario/FactoriaMontana.rb
require('./S1/FactoriaCarreraYBicicleta.rb');

module S1
	public
	class FactoriaMontana < FactoriaCarreraYBicicleta

		public
		def crearBicicleta(valor)
			BiciMontana.clone(valor)
		end

		public
		def crearCarrera
			CarreraMontana.clone
		end
	end
end
